include/core/io/platform/f4xx/ADCf4xx.hpp

```diff
@@ -25,22 +32,79 @@ class ADCf4xx : public ADC {
-    static ADC_HandleTypeDef halADC;
-    /// Static list of all channels supported by the ADC
-    static Pin channels[MAX_CHANNELS];
-    /// Buffer for DMA where each spot represents the value read in from a
-    /// channel
-    static uint16_t buffer[MAX_CHANNELS];
-    static DMA_HandleTypeDef halDMA;
+    // Flag to indicate if the timer has been initialized
+    static bool timerInit;
+    // Timer handle for TIM8, used to configure and control the timer instance
+    static TIM_HandleTypeDef htim8;
+
+    /**
+     * Structure to represent the state of an ADC instance.
+     *
+     * This structure holds the configuration and current state for an ADC instance, including the
+     * HAL handle for the ADC and DMA, channel configurations, and data buffers for readings.
+     *
+     * halADC: HAL handle for configuring and controlling the ADC peripheral.
+     * rank: The ADC channel rank in the sequence (starts at 1).
+     * isADCInit: Flag to indicate whether the ADC has been initialized.
+     * channels: Array of pins mapped to ADC channels. Initialized to all Pin::DUMMY.
+     *             The array size is defined by MAX_CHANNELS.
+     * buffer: Array to store ADC values for each channel, indexed according to the channels array.
+     *           Each entry corresponds to a specific ADC channel.
+     * halDMA: HAL handle for configuring and controlling DMA for the ADC.
+     */
+    struct ADC_State_t {
+        ADC_HandleTypeDef halADC      = {0};
+        uint8_t rank                  = 1;
+        bool isADCInit                = false;
+        Pin channels[MAX_CHANNELS]    = {Pin::DUMMY,
+                                         Pin::DUMMY,
+                                         Pin::DUMMY,
+                                         Pin::DUMMY,
+                                         Pin::DUMMY,
+                                         Pin::DUMMY,
+                                         Pin::DUMMY,
+                                         Pin::DUMMY,
+                                         Pin::DUMMY,
+                                         Pin::DUMMY,
+                                         Pin::DUMMY,
+                                         Pin::DUMMY,
+                                         Pin::DUMMY,
+                                         Pin::DUMMY,
+                                         Pin::DUMMY,
+                                         Pin::DUMMY};
+        uint16_t buffer[MAX_CHANNELS] = {0};
+        DMA_HandleTypeDef halDMA      = {0};
+    };
+
+    // Array of all ADC peripheral states
+    static ADC_State_t adcArray[NUM_ADCS];
+    // The ADC peripheral state of the current object
+    ADC_State_t& adcState;
+    // The number ADC peripheral which is being used in the current object
+    uint8_t adcNum;
+
+    /**
+     * Checks if the channel that is being initialized supports the ADC peripheral that it is being initialized on.
```

Comment:
Function summaries should be written as commands
```suggestion
     * Check if the channel that is being initialized supports the ADC peripheral that it is being initialized on.
```